### ISSUE TITLE
Document current limitations of datapipes

### ISSUE THREAD

**Issue by NicolasHug** [pytorch/text#1751]
As discussed with @parmeet previously, it's important to document current limitations of the datasets. In particular regarding:

-  the lack of DDP support
- how to properly shuffle (i.e. pass shuffle=True in the DataLoader). We should also make sure that suffle works as expected regarding RNGs, which was broken benfore. There were recent fixes in the DataLoader regarding this - @parmeet did you get a chance to verify that things work as expected now?

As discussed, the best way to document this is probably to add a `..warning::` note in all of the datasets docstrings, or alternatively, if the warning is too long, to write one long warning on top of the datasets.rst page, and to write smaller warnings in all docstrings that link to this bigger one.

Ideally this should happen before the release cut. @parmeet, LMK if you'd like me to submit a PR

**Comment by Nayef211** [comment]
@NicolasHug thank you for calling this out. I do like the second option of writing one warning on top of the `datasets.rst` page and having warnings in each dataset linking to the main warning just to avoid duplicating a large warning message across all our dataset implementations. Would you mind creating a PR that calls out some of these limitations since you have a lot of context here and brought up many of these issues to light? 😄
